datadog_checks_base/datadog_checks/utils/common.py

```diff
@@ -2,9 +2,35 @@
 # All rights reserved
 # Licensed under a 3-clause BSD style license (see LICENSE)
 import os
+import re
 
 from six.moves.urllib.parse import urlparse
 
 
 def get_docker_hostname():
     return urlparse(os.getenv('DOCKER_HOST', '')).hostname or 'localhost'
+
+
+def pattern_filter(items, patterns, exclude=False, key=None):
```

Comment:
Let's change the logic so that the function takes a `whitelist` and a `blacklist` sequence. This way we can set a single source of truth in terms of what takes precedence when an integration config supports both white and black lists.
For example, let's say an integration has something like:
```
instances:
 - foo: bar
   blacklist_projects: 
     r'staging.+'
     r'test_.+'
   whitelist_projects: 
     'staging_prod_candidate'
```
Would `staging_prod_candidate` be included or not by the filter? (I think it should be included).

Reply:
I think this would be counterproductive to having a generalized filter function; it would be wrong to dictate the priority for every integration. If an integration happens to have both types of lists (extremely rare) it can simply chain this function in the order most appropriate for it.

Reply:
I think I agree with being able to handle both whitelist and blacklists in the same function. 

Or if we wanted to create a wrapper function around this that took both a whitelist and blacklist and chained this for you. That way we can have a `default` behavior while allowing the flexibility to modify it if an integration needs to. 

What do you think?

Reply:
I'll do a wrapper.

Reply:
I added our default chaining function.

Reply:
`pattern_filter_chain ` does all we need, let's rename it to `pattern_filter` and we're good

Reply:
Done! Also, I made individual functions for performance reasons.